jina/executors/encoders/audio/spectral.py

```diff
@@ -0,0 +1,89 @@
+import numpy as np
+
+from .. import BaseAudioEncoder
+from ...decorators import batching, as_ndarray
+
+
+class MFCCTimbreEncoder(BaseAudioEncoder):
+    """
+    :class:`MFCCTimbreEncoder` is based on Mel-Frequency Cepstral Coefficients (MFCCs) which represent timbral features.
+    :class:`MFCCTimbreEncoder` encodes an audio signal from a `Batch x Signal Length` ndarray into a
+    `Batch x Concatenated Features` ndarray.
+    """
+
+    def __init__(self, input_sample_rate: int = 22050, n_mfcc=20, n_fft_length=2048, hop_length: int = 512, *args,
+                 **kwargs):
+        """
+        :class:`MFCCTimbreEncoder` extracts from an audio signal `n_mfcc`-dimensional feature vectors and concatenate
+        them.
+
+        :param input_sample_rate: input sampling rate in Hz (22050 by default)
+        :param n_mfcc: the number of coefficients (20 by default)
+        :param n_fft: length of the FFT window (2048 by default)
+        :param hop_length: the number of samples between successive MFCC frames (512 by default)
+        """
+        super().__init__(*args, **kwargs)
+        self.input_sample_rate = input_sample_rate
+        self.n_mfcc = n_mfcc
+        self.n_fft_length = n_fft_length
+        self.hop_length = hop_length
+
+    @batching
+    @as_ndarray
+    def encode(self, data: np.ndarray, *args, **kwargs) -> np.ndarray:
+        """
+        Segments the audio signal of each Chunk into short MFCC frames, extracts MFCCs for each frame and concatenates
+        Chunk frame MFCCs into a single Chunk embedding.
+
+        :param data: a `Batch x Signal Length` ndarray, where `Signal Length` is a number of samples
+        :return: a `Batch x Concatenated Features` ndarray, where `Concatinated Features` is a `n_mfcc`-dimensional
+        feature vector times the number of the MFCC frames
+        """
+        import librosa
```

Comment:
```suggestion
        from librosa.feature import mfcc
```
Importing the single function will save the indexing time and avoid the overhead of importing the whole lib.